### PR TITLE
Remove NetMessage deprecated boilerplate entirely

### PR DESCRIPTION
**Pull request by mirrorcult**
Removes all boilerplate/obsoleted uses of netmessage constructors/nameof stuff deprecated in #1770

Requires space-wizards/space-station-14#4196

### CODE REVIEW

File: Robust.UnitTesting/RobustIntegrationTest.NetManager.cs [hunk 1]
```diff
@@ -272,21 +272,16 @@ public void ServerSendToMany(NetMessage message, List<INetChannel> recipients)
             public event EventHandler<NetChannelArgs>? Connected;
             public event EventHandler<NetDisconnectedArgs>? Disconnect;
 
-            public void RegisterNetMessage<T>(string name, ProcessMessage<T>? rxCallback = null,
-                NetMessageAccept accept = NetMessageAccept.Both) where T : NetMessage
+            public void RegisterNetMessage<T>(ProcessMessage<T>? rxCallback = null, NetMessageAccept accept = NetMessageAccept.Both) where T : NetMessage, new()
             {
+                var name = new T().MsgName;
```
Comment (metalgearsloth) [inline Robust.UnitTesting/RobustIntegrationTest.NetManager.cs:277]: Is this used?

File: Robust.Shared/Network/NetManager.cs [hunk 2]
```diff
@@ -902,10 +902,11 @@ private void CacheNetMsgFunction(byte id)
         #region NetMessages
 
         /// <inheritdoc />
-        public void RegisterNetMessage<T>(string name, ProcessMessage<T>? rxCallback = null,
+        public void RegisterNetMessage<T>(ProcessMessage<T>? rxCallback = null,
             NetMessageAccept accept = NetMessageAccept.Both)
-            where T : NetMessage
+            where T : NetMessage, new()
         {
+            var name = new T().MsgName;
```
Comment (metalgearsloth) [inline Robust.Shared/Network/NetManager.cs:909]: Won't this allocate a new message?

File: Robust.UnitTesting/RobustIntegrationTest.NetManager.cs (hunk 1 above)
Comment (mirrorcult) [reply Robust.UnitTesting/RobustIntegrationTest.NetManager.cs:277]: guess not, might have to update tests to use it?

File: Robust.Shared/Network/NetManager.cs (hunk 2 above)
Comment (mirrorcult) [reply Robust.Shared/Network/NetManager.cs:909]: idk, this is how it was done before, just moved into this function

File: Robust.UnitTesting/RobustIntegrationTest.NetManager.cs (hunk 1 above)
Comment (metalgearsloth) [reply Robust.UnitTesting/RobustIntegrationTest.NetManager.cs:277]: Maybe just comment it out and leave a note explaining that it isn't used here but is used in the live one or something?

**Pull request merged**